pytorch_fairseq_translation.md

```diff
@@ -0,0 +1,122 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+title: Transformer (NMT)
+summary: Transformer models for English-French and English-German translation.
+category: researchers
+image: fairseq_logo.png
+author: Facebook AI Research (fairseq Team)
+tags: [nlp]
+github-link: https://github.com/pytorch/fairseq/
+accelerator: cuda-optional
+order: 10
+---
+
+
+### Model Description
+
+The Transformer, introduced in the paper [Attention Is All You Need][1], is a
+powerful sequence-to-sequence modeling architecture capable of producing
+state-of-the-art neural machine translation (NMT) systems.
+
+Recently, the fairseq team has explored large-scale semi-supervised training of
+Transformers using back-translated data, further improving translation quality
+over the original model. More details can be found in [this blog post][2].
+
+
+### Requirements
+
+We require a few additional Python dependencies for preprocessing:
+
+```bash
+pip install regex requests sacremoses sentencepiece subword_nmt
+```
+
+
+### English-to-French Translation
+
+To translate from English to French using the model from the paper [Scaling
+Neural Machine Translation][3]:
+
+```python
+import torch
+en2fr = torch.hub.load(
+  'pytorch/fairseq',
+  'transformer',
+  model_name_or_path='transformer.wmt14.en-fr',
+  data_name_or_path='.',
+  tokenizer='moses',
+  aggressive_dash_splits=True,
+  bpe='subword_nmt',
+)
+print(en2fr.generate('Hello world!'))
+# Bonjour à tous !
+```
```

Comment:
Maybe we should also demonstrate how to access the underlying nn.Module model itself (or at least mention that it is available)? Its not immediately clear that this is accessible from the torch hub output.